client/components/clip-list-item.js

```diff
@@ -23,23 +24,25 @@ const classNames = {
 const template = html`
 	<img class="${classNames.THUMBNAIL}" src="data:," alt="" />
 	<span class="${classNames.LABEL}"></span>
+	<span class="${classNames.DURATION}"></span>
 `
 
 class ClipListElement extends HTMLLIElement {
 	constructor() {
 		super()
 
 		this.classList.add(classNames.ROOT)
-		this.innerHTML = template
+		this.insertAdjacentHTML('afterbegin', template)
```

Comment:
Note to @gundelsby : I've found on [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Element/insertAdjacentHTML) that `.insertAdjacentHTML()` is faster than `.innerHTML`, so I switched to using that.

Reply:
Very good 👍
Just remember to verify that you're working on empty (new) nodes (like here), since it obviously inserts content, as opposed to replacing (which is what `element.innerHTML = whatever` does)